dradis/dradis-pdf_export
# Enable templates

**Spec**
Our docs: https://dradisframework.com/support/guides/reporting/pdf_reports.html#custom-ce say that using this add-on we should be able to use a `.rb` file as a template, so we can manipulate the default pdf report.
But that is not working.

**Proposed Solution**
The PDF is generated using the `prawn` gem in our `Processor` class. 
In this PR we check if we have a template defined the redefines that class.
If the template exists we use `load` to use the `Processor` class defined by the template.
If the template dos not exist we use `load` to use the already present `Processor` class.
Since we only want to overwrite the `Processor` class in the templates, we extract that class to its own file in this PR.

**NOTE:** If this PR is accepted, we must update the linked docs above so the file used to create templates is `processor.rb` and not `exporter.rb`

**How to test**

- Enable the add-on by adding something like this in your `Gemfile.plugins`:
`gem 'dradis-pdf_export', path: '../dradis-pdf_export'` (checkout `fix-templates` branch)
- Generate a pdf report. Assert it contains dradis logo at the beginning.
- Copy the `processor.rb` file as `edited.rb` in templates folder (`dradis-ce/templates/reports/pdf_export/` ?). In that file comment around L21 the call to `cover_page`
- Generate the PDF report again and assert that no logo appears at the beginning
- Delete the template, regenerate the PDF report, assert the logo is back.
